allow build.b_* options

dcbaker opened the pull request:
These continue to be ignored as they always have, but no longer raise an
error.

Fixes: #8351
Fixes: #8354

jpakkane commented:
Now that we have native files it might make sense to prohibit setting build machine options in cross files (if I had realized that I would have made it an error when the feature was originally created). We can't break this immediately, but maybe we could print a deprecation warning that this might go away at some point.

dcbaker commented:
I'm proposing removing it in 0.60, since the correct solution has always worked, but I've fixed it to work correctly now, and added the deprecation warning.

jpakkane commented:
Test failures seem relevant.

dcbaker commented:
@jpakkane I have set the deprecation warning to mention 0.60 as the removal target, since the "right way" has always been available. Is that okay, or do you want me to change that?

jpakkane commented:
Please rebase. It should go green.
